src/lib.rs

```diff
@@ -490,3 +513,48 @@ impl Builder {
             .expect("Builder::init should not be called after logger initialized")
     }
 }
+
+/// Construct a log entry and send it to the logd writer socket
+///
+/// This can be used to forge an android logd entry
+///
+/// # Example
+///
+/// ```
+/// use android_logd_logger::{log, Buffer, Priority};
+/// use chrono::prelude::Utc;
+///
+/// let time = Utc::now();
+///
+/// log(Buffer::Main, 0, 0, time.timestamp() as u32, time.timestamp_subsec_nanos(),
+/// &tag, Priority::Info, &log)
+/// ```
+pub fn log(
+    buffer_id: Buffer,
+    pid: u16,
+    thread_id: u16,
+    timestamp_secs: u32,
+    timestamp_subsec_nanos: u32,
+    tag: &str,
+    priority: Priority,
+    message: &str,
+) -> Result<(), Error> {
+    let record = Record {
+        timestamp_secs,
+        timestamp_subsec_nanos,
+        pid,
+        thread_id,
+        buffer_id,
+        tag,
+        priority,
+        message,
+    };
+
+    #[cfg(target_os = "android")]
+    crate::logd::log(&record);
+
+    #[cfg(not(target_os = "android"))]
+    println!("buffer: {:?}, tag: {:?}, message: {:?}", buffer_id, tag, message);
```

Comment:
Please forward the records in the not android case as well. This keeps the equals format.